src/bin/ion/commands/beta/generate/context.rs

```diff
@@ -0,0 +1,60 @@
+use crate::commands::beta::generate::utils::Language;
+use serde::Serialize;
+use tera::Tera;
+
+/// Represents a context that will be used for code generation
+pub struct CodeGenContext {
+    // Represents the templating engine - tera
+    pub(crate) tera: Tera,
+    pub(crate) language: Language,
+    // Initially the data_model field is set to None.
+    // Once an ISL type definition is mapped to a data model this will have Some value.
+    pub(crate) data_model: Option<DataModel>,
+    // Represents a counter for naming anonymous type definitions
+    pub(crate) anonymous_type_counter: usize,
+}
+
+impl CodeGenContext {
+    pub fn new(language: Language) -> Self {
+        Self {
+            language,
+            data_model: None,
+            anonymous_type_counter: 0,
+            tera: Tera::new("src/bin/ion/commands/beta/generate/templates/**/*.templ").unwrap(),
+        }
+    }
+
+    pub fn with_data_model(&mut self, data_model: DataModel) {
+        self.data_model = Some(data_model);
+    }
+
+    pub fn with_initial_data_model(&mut self) {
+        // Initially the data model is set to None, this will be set with Some(_) value when data model is determined in code generation process
+        self.data_model = None;
+    }
+
+    /// Returns a string that represent the template name based on data model type.
+    pub fn template_name(&self) -> &str {
+        if let Some(data_model) = &self.data_model {
+            return match (&self.language, data_model) {
+                (
+                    Language::Rust,
+                    DataModel::Struct | DataModel::UnitStruct | DataModel::SequenceStruct,
+                ) => "struct",
+                (
+                    Language::Java,
+                    DataModel::Struct | DataModel::UnitStruct | DataModel::SequenceStruct,
+                ) => "class",
+            };
+        }
+        "" // Default value is an empty string
```

Comment:
Like I mentioned in a previous comment, this name (template file name) will be used to get the template file for generate code. (e.g. struct.templ)
Hence a default value would result in an error from tera templating engine as it can't find a template with that empty string name. (currently only mod.templ and struct.templ exist)

Reply:
Got it. Since others might also be confused by this, I suggest making this method return a `Result` so this path leads to an explicit error. As-is it's not obvious that getting the default value back is actually a problem.